ComSemApp/templates/ComSemApp/student/assessment.html

```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
   });
 }
 
+function timer_promise(sentence){
+  let reading_check = timer(sentence.length/2, READING_OFFSET);
+  //Once reading timer done this starts
+  reading_check.then(() => {
+    $("#timer-bar").removeClass("bg-info").addClass("bg-danger")
+    //Runs the recording timer
+
+    //starts recording
+    recorder.start();
+    let recording_check = timer(sentence.length/2, RECORDING_OFFSET);
+    //Once recording timer done this starts
+    recording_check.then(() => { 
+      // stops recording
+      recorder.stop();
+      // keeps next problem button disabled if no more problems left
+      if (curr_problem_index < problems.length -1 ){
+        toggle_enabled_next();
+      }
+      
+
+      // updates the progress bar
+      process_response();
+      save_in_form();
```

Comment:
This save_in_form call needs to be dependent on the recorder's dataAvailable event because audioBlob will not be assigned until then.